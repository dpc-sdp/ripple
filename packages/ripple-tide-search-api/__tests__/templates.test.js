import coreTemplates from './../services/templates'

describe('Tide Search API built in templates', () => {
  test('should have site template', async () => {
    expect(coreTemplates).toHaveProperty('site')
    expect(coreTemplates.site).toHaveProperty('requestMapping')
    expect(typeof coreTemplates.site.requestMapping).toEqual('function')
  })

  test('should return basic request if no params sent', async () => {
    expect(coreTemplates.site.requestMapping()).toEqual({
      query: { match_all: {} }
    })
  })

  test('should return correct search query', async () => {
    const params = {
      q: 'test'
    }
    expect(coreTemplates.site.requestMapping(params)).toMatchSnapshot()
  })

  test('set type filter', async () => {
    const params = {
      type: 'test'
    }
    expect(coreTemplates.site.requestMapping(params)).toMatchSnapshot()
  })

  describe('filter', () => {
    test('should return correct terms filters', async () => {
      const params = {
        filters: {
          type: {
            values: ['event']
          }
        }
      }
      expect(coreTemplates.site.requestMapping(params)).toMatchSnapshot()
    })

    test('should return correct term filters', async () => {
      const params = {
        filters: {
          type: {
            values: 'event'
          }
        }
      }
      expect(coreTemplates.site.requestMapping(params)).toMatchSnapshot()
    })
  })

  describe('pagination', () => {
    test('should return size', async () => {
      const query = coreTemplates.site.requestMapping({
        limit: 9,
        page: 1
      })
      expect(query.size).toEqual(9)
    })

    test('should return from on page 2', async () => {
      const query = coreTemplates.site.requestMapping({
        limit: 4,
        page: 2
      })
      expect(query.size).toEqual(4)
      expect(query.from).toEqual(4)
    })

    test('should return paginated response on page 3', async () => {
      const query = coreTemplates.site.requestMapping({
        limit: 4,
        page: 3
      })
      expect(query.size).toEqual(4)
      expect(query.from).toEqual(8)
    })
  })
})
