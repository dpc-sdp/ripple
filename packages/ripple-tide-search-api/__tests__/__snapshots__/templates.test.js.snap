// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tide Search API built in templates filter should return correct term filters 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "match_all": Object {},
    },
    "filter": Object {
      "bool": Object {
        "must": Array [
          Object {
            "term": Object {
              "type": "event",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Tide Search API built in templates filter should return correct terms filters 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "match_all": Object {},
    },
    "filter": Object {
      "bool": Object {
        "must": Array [
          Object {
            "terms": Object {
              "type": Array [
                "event",
              ],
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Tide Search API built in templates set type filter 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "match_all": Object {},
    },
    "filter": Object {
      "bool": Object {
        "must": Array [
          Object {
            "term": Object {
              "type": "test",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Tide Search API built in templates should return correct search query 1`] = `
Object {
  "query": Object {
    "bool": Object {
      "must": Object {
        "multi_match": Object {
          "fields": Array [
            "body",
            "field_landing_page_summary",
            "field_paragraph_body",
            "field_paragraph_summary",
            "title",
          ],
          "query": "test",
        },
      },
      "should": Object {
        "match_phrase": Object {
          "title": Object {
            "boost": 2,
            "query": "test",
          },
        },
      },
    },
  },
}
`;
